src/interface/variable_pack.hpp

```diff
@@ -105,8 +117,9 @@ class VariableFluxPack : public VariablePack<T> {
   VariableFluxPack() = default;
   VariableFluxPack(const ViewOfParArrays<T> view, const ViewOfParArrays<T> f0,
                    const ViewOfParArrays<T> f1, const ViewOfParArrays<T> f2,
-                   const ParArray1D<int> sparse_ids, const std::array<int, 4> dims)
-      : VariablePack<T>(view, sparse_ids, dims), f_({f0, f1, f2}) {}
+                   const ParArray1D<int> sparse_ids,
+                   const ParArray1D<int> vector_component, const std::array<int, 4> dims)
```

Comment:
The same I think is true here, it looks like the arguments are being copied when they are passed in and also when they are initialized.